File: alica_engine/src/engine/modelmanagement/factories/AbstractPlanFactory.cpp
#include "engine/modelmanagement/factories/AbstractPlanFactory.h"
#include "engine/model/AbstractPlan.h"
#include "engine/modelmanagement/Strings.h"
#include "engine/modelmanagement/factories/VariableFactory.h"

namespace alica
{
void AbstractPlanFactory::setVariables(const YAML::Node& abstractPlanNode, alica::AbstractPlan* abstractPlan)
{
    if (Factory::isValid(abstractPlanNode[alica::Strings::variables])) {
        const YAML::Node& variableNodes = abstractPlanNode[alica::Strings::variables];
        for (YAML::const_iterator it = variableNodes.begin(); it != variableNodes.end(); ++it) {
            abstractPlan->addVariable(VariableFactory::create(*it));
        }
    }
}
} // namespace alica

File: alica-supplementary/plan-conversion-tool/src/main/java/com/rapyutarobotics/alica/factories/AbstractPlanFactory.java
package com.rapyutarobotics.alica.factories;

import com.rapyutarobotics.alica.ConversionProcess;
import com.rapyutarobotics.alica.Tags;
import de.unikassel.vs.alica.planDesigner.alicamodel.AbstractPlan;
import org.w3c.dom.Element;
import org.w3c.dom.NodeList;

public class AbstractPlanFactory extends Factory {

    public static void setVariables(Element node, AbstractPlan abstractPlan, ConversionProcess cp)
    {
        NodeList listOfVariableNodes = node.getElementsByTagName(Tags.VARIABLES);
        for (int i = 0; i < listOfVariableNodes.getLength(); i++) {
            Element variableNode = (Element) listOfVariableNodes.item(i);
            if (variableNode.getParentNode().getNodeName().equals(Tags.PLANTAG)) {
                // filter for those <vars..> nodes that are direct children of <alica:plan..> nodes
                abstractPlan.addVariable(VariableFactory.create(variableNode, cp));
            }
        }
    }
}
